cgi-bin/LJ/Talk.pm

```diff
@@ -3695,7 +3695,13 @@ sub make_preview {
             unless $_ eq 'body' || $_ eq 'subject' || $_ eq 'prop_opt_preformatted' || $_ eq 'editreason';
     }
 
-    $ret .= "<br /><input type='submit' value='$BML::ML{'/talkpost_do.bml.preview.postcomment'}' />\n";
+    my $post_disabled = $u->does_not_allow_comments_from($remote)
+        || ($u->{'opt_whocanreply'} eq 'reg' && $remote->is_identity && ! ( $remote->is_validated || $u->trusts( $remote )));
```

Comment:
Okay, so then we can't fold it into that one method, but we could at least abstract this set of logic into a new method? That shouldn't break anything, and at least keeps this code simpler (and ensures that anybody who touches either of these places in the future understands they're related).